Issue #29: Just splitting db server into another module

ekes opened the pull request.

ekes committed:
Start splitting up.

Make search index disabled with no server, minimal configuration.
Plan make _db _solr modules augment and enable (if not already there).

ekes committed:
Automatically enable index with server enabled.

Processors and any other settings should still be added either
automatically with the install hook change, or with some UI.
Question https://github.com/localgovdrupal/localgov_search/issues/29#issuecomment-1175221402

rupertj committed:
Moved view and display ids to be properties of the SitewideSearchBlock plugin, so they can be changed if this class is extended.

ekes committed:
Going to have to review best way to add in this config if we want to tweak it.

ekes committed:
Add uninstall and tests.

ekes committed:
Get solr test to pass.

ekes committed:
CS: unused use statement.

ekes committed:
Update composer for search_api_solr.

ekes committed:
The resturn of the solr config.

Some of these are _required_ when installing with an install profile.
But these can't be in config/install because when installing after
profile they confict.

Might well also be the place to put tweaks in, but need to make sure
they are overriding the solr module configuration.

ekes committed:
More config shenanigans for install profile.

ekes committed:
Set our default processors on existing index on installation.

ekes committed:
And setting the default processors for database.

rupertj committed:
Fixes code standards issue.

ekes committed:
Updated default lando solr server and core to match https://github.com/localgovdrupal/localgov_project/pull/87/commits/ecb8819d51101406c629a2c4aadd79099162d752

ekes committed:
Remove localgov_search_solr module to be added in seperate PR.

finnlewis commented:
Would be good to get some other eyes on this before merging.

finnlewis commented:
@andybroomfield @stephen-cox @Adnan-cds it woudl be great to get some more eyes on this, especially on existing installed sites. 

This will free up enabling different backends.

finnlewis commented:
This is linked to https://github.com/localgovdrupal/localgov_directories/pull/206

finnlewis commented:
From discussion in the search working group: 

@ekes thinks it would be good to write an update path to enable the module but not install it's config! 

@ekes will need to look at this before further testing.

ekes committed:
Enable _db on existing installs; and clarify description.

It is 'as if' the _db module was used for installing configuration on
existing sites, so making that clear by enabling, but bypassing
configuration.

The description clarifies the module provides the initial server
configuration.

Adnan-cds commented on localgov_search.install:
```diff
@@ -16,3 +16,16 @@ function localgov_search_install() {
 }
+
+/**
+ * Sitewide Search Search API database becomes its own module.
```
I am finding this description confusing :(

ekes replied:
Updated

Adnan-cds commented on modules/localgov_search_db/localgov_search_db.install:
```diff
@@ -0,0 +1,45 @@
+    $server = Server::load('localgov_sitewide_search');
+    $module_path = \Drupal::service('extension.list.module')->getPath('localgov_search_db');
+    $index_solr_config = Yaml::decode(file_get_contents($module_path . '/config/conditional/search_api.index.localgov_sitewide_search.yml'));
+    $new_config_index = Index::create($index_solr_config);
```
The variable name `$index_solr_config` looks out of context here.

ekes replied:
Corrected

Adnan-cds reviewed (approved):
Works for me and code looks mostly okay.  I have pointed out two minor issues.

I noticed that the search block is available on a fresh LocalGov site with the localgov_search module enabled and the localgov_search_db submodule disabled.  This is an unlikely scenario in real-life use.  So not a big deal but good to be aware of.

ekes committed:
Fixes to suggestions for clarification.

ekes committed:
Rather than show search box when disabled, explain needs backend.

ekes commented:
> I noticed that the search block is available on a fresh LocalGov site with the localgov_search module enabled and the localgov_search_db submodule disabled.

What about https://github.com/localgovdrupal/localgov_search/pull/40/commits/32fd1d26ec35aec82361525cfeaae35ebdc4af52

ekes committed:
Merge remote-tracking branch 'origin/1.x' into feature/1.x/29-db

finnlewis commented:
Discussion in Merge Monday. 
To anyone with an existing site, this change should have no noticeable effect.
People with current fresh builds might need to change the config of their install process to include the new sub-module.
Adnan has reviewed and approved. 
Version number will bump the minor version to 1.1.0
So any objections to merging and doing a release?  None.
